transport/http/transport.go

```diff
@@ -200,6 +201,22 @@ func Logger(logger *zap.Logger) TransportOption {
 	}
 }
 
+// DialerCalled is used for tests to see whether we reached a certain point in tests. Currently we check
+// if we caught an error from Dialer.
+func DialerCalled(dialerCalled func()) TransportOption {
+	return func(options *transportOptions) {
+		options.dialerCalled = dialerCalled
+	}
+}
+
+// CloserCalled is used for tests to see whether we reached a certain point in tests. Currently we check
+// if we caught an error from Dialer.
+func CloserCalled(closerCalled func()) TransportOption {
+	return func(options *transportOptions) {
```

Comment:
APIs needed exclusively for testing can be placed in a `transport_test.go` file. These
should become available as exported functions to other tests.

Although in this case, even that isn't needed because the functions are used
for testing from within the same package.